src/bin/help.ts

```diff
@@ -0,0 +1,172 @@
+		| "core"
+		| "offline"
+		| "opt-out"
+		| "optional"
+		| "skip-disk"
+		| "skip-net";
+	multiple?: boolean;
+	type: string;
+}
+
+function logHelpTextSection(section: HelpTextSection): void {
+	console.log(" ");
+
+	console.log(`   ${chalk.black.bgGreenBright(section.sectionHeading)}`);
+
+	for (const subsection of section.subsections) {
+		if (subsection.warning) {
+			console.log(chalk.yellow(subsection.warning));
+		}
+
+		if (subsection.subheading) {
+			console.log(chalk.green(subsection.subheading));
+		}
+
+		for (const option of subsection.flags) {
+			const { description, flag, type } = option;
+			console.log(
+				chalk.cyan(
+					`
+      --${flag}${
+				type !== "boolean" ? ` (${chalk.cyanBright(type)})` : ""
+			}: ${description}`,
+				),
+			);
+		}
+	}
+}
+
+function createHelpTextSections(
+	options: Record<string, Option>,
+): HelpTextSection[] {
+	const core: HelpTextSection = {
+		sectionHeading: "Core options:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: undefined,
+			},
+		],
+	};
+
+	const optional: HelpTextSection = {
+		sectionHeading: "Optional options:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: undefined,
+			},
+		],
+	};
+
+	const optOut: HelpTextSection = {
+		sectionHeading: "Opt-outs:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: `
+      ⚠️ Warning: Specifying any --exclude-* flag on the command-line will 
+      cause the setup script to skip prompting for more excludes. ⚠️`,
+			},
+			{
+				flags: [
+					{
+						description: `Skips network calls that fetch all-contributors
+        data from GitHub`,
+						flag: "exclude-contributors",
+						type: "boolean",
+					},
+				],
+				subheading: `
+      You can prevent the migration script from making some network-based 
+      changes using any or all of the following CLI flags:`,
+				warning: undefined,
+			},
+			{
+				flags: [],
+				subheading: `
+      You can prevent the migration script from making some changes on disk 
+      using any or all of the following CLI flags:`,
+				warning: undefined,
+			},
+		],
+	};
+
+	const offline: HelpTextSection = {
+		sectionHeading: "Offline Mode:",
+		subsections: [
+			{
+				flags: [],
+				subheading: undefined,
+				warning: undefined,
+			},
+		],
+	};
+
+	const subsections = {
+		core: core.subsections[0],
+		offline: offline.subsections[0],
+		"opt-out": optOut.subsections[0],
+		optional: optional.subsections[0],
+		"skip-disk": optOut.subsections[2],
+		"skip-net": optOut.subsections[1],
+	};
+
+	for (const [option, data] of Object.entries(options)) {
+		subsections[data.docsSection].flags.push({
+			description: data.description,
+			flag: option,
+			type: data.type,
+		});
+	}
+
+	return [core, optional, optOut, offline];
+}
+
+export function logHelpText(): void {
+	const helpTextSections = createHelpTextSections(
+		allArgOptions as Record<string, Option>,
```

Comment:
[Cleanup] Nit: seems to be unnecessary? Nothing breaks in `pnpm tsc --watch` when I remove it.

```suggestion
		allArgOptions,
```